<launch>

    <!-- Mobile base node -->
    <node name="base" pkg="aidu_mobile_base" type="base" output="screen" respawn="false">
        <!-- launch-prefix="gdb -ex run - -args" -->
        <param name="motor_port" value="motor_comm"/>
        <param name="motor_config" value="$(find aidu_mobile_base)/config/mobile_base.xml"/>
        <param name="wheel_diameter" value="0.295"/>
        <param name="wheel_base" value="0.54"/>
    </node>

    <!-- Shared serial communication with threemxl -->
    <node name="motor_comm" pkg="shared_serial" type="server" output="screen">
        <param name="port_name" value="/dev/ttyUSB0"/>
        <param name="port_type" value="RS485_FTDI"/>
        <param name="baud_rate" value="921600"/>
    </node>
    
    <!-- Odometry node -->
    <node name="odometry" pkg="aidu_mobile_base" type="odom" output="screen" />

</launch>
